iiod/responder.c

```diff
@@ -715,6 +756,10 @@ static void handle_free_block(struct parser_pdata *pdata,
 		if (entry->block != block)
 			continue;
 
+		ep_fd = buf_entry->is_tx ? pdata->fd_in : pdata->fd_out;
+		if (WITH_IIOD_USB_DMABUF && entry->dmabuf_fd > 0)
+			usb_detach_dmabuf(ep_fd, entry->dmabuf_fd);
```

Comment:
nit: another place where we could already have some "backend agnostic" mechanism in place...

Reply:
But this is IIOD and not Libiio, and interfaces gadget/functionfs and not client-side USB, so it wouldn't fit in one of Libiio's backends.

Reply:
I'm aware it's IIOD, but for example if we ever get support for DMABUF in network (I know this is a big 'if') would we have something like `net_detach_dmabuf()`? So wouldn't we need to have another another:

```
if (WITH_IIOD_NET_DMABUF && ...)
   ...
```

But maybe there's no `ops` kind of thing in IIOD and adding new helpers for this, yeah, can be done when really needed... 

Or maybe I'm just completely off here and should just check iiod code :sweat_smile:

Reply:
Yes, there are no "frontends" in IIOD yet, and adding this separation would be a bit outside the scope of this PR. I would refactor this the moment where we have DMABUF support on the network stack.